pkg/services/ngalert/store/instance_database.go

```diff
@@ -149,11 +177,18 @@ func (st DBstore) FetchOrgIds(ctx context.Context) ([]int64, error) {
 	return orgIds, err
 }
 
-func (st DBstore) DeleteAlertInstance(ctx context.Context, orgID int64, ruleUID, labelsHash string) error {
+func (st DBstore) DeleteAlertInstances(ctx context.Context, keys ...InstanceKey) error {
+	if len(keys) == 0 {
+		return nil
+	}
+
 	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		_, err := sess.Exec("DELETE FROM alert_instance WHERE rule_org_id = ? AND rule_uid = ? AND labels_hash = ?", orgID, ruleUID, labelsHash)
-		if err != nil {
-			return err
+		for _, k := range keys {
```

Comment:
TODO: should probably use a row limit here as well, so our transactions don't get too big.